proxy.go

```diff
@@ -41,22 +41,27 @@ func NewBucketingProxyInstance(instance *ProxyInstance) (*ProxyInstance, error)
 			return nil, fmt.Errorf("HTTP port must be set")
 		}
 		go func() {
-			err := r.Run(":" + strconv.Itoa(instance.HTTPPort))
+			err = r.Run(":" + strconv.Itoa(instance.HTTPPort))
 			if err != nil {
 				log.Printf("Error running HTTP server: %s", err)
 			}
 		}()
 		log.Printf("HTTP server started on port %d", instance.HTTPPort)
 	}
 	if instance.UnixSocketEnabled {
-		if _, err := os.Stat(instance.UnixSocketPath); err == nil {
+		if _, err = os.Stat(instance.UnixSocketPath); err == nil {
 			return nil, fmt.Errorf("unix socket path %s already exists. Skipping instance creation", instance.UnixSocketPath)
 		}
+		err = nil
 		go func() {
-			err := r.RunUnix(instance.UnixSocketPath)
+			err = r.RunUnix(instance.UnixSocketPath)
 			if err != nil {
 				log.Printf("Error running Unix socket server: %s", err)
 			}
+			fileMode := os.FileMode(instance.UnixSocketPermissions)
+			if err = os.Chmod(instance.UnixSocketPath, fileMode); err != nil {
+				log.Printf("Error setting Unix socket permissions: %s", err)
+			}
```

Comment:
I tried running this locally, and it doesn't seem to run. I'm guessing that `r.RunUnix` is a blocking call, and it only gets here when the server shuts down.
You might have to move the creation of the socket out of there so that it can follow this order:
1. create socket
2. chmod socket
3. run server

Reply:
hmm - that's true - i might have to move this outside of the goroutine